packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1290,7 +1290,9 @@ function collectRecommendations({
       if (child.isSatisfiedRecursively) {
         if (child.hasRequiredNodesBelow) {
           // Remember this dep actually satisfied something.
-          satisfyingPaths.add(path);
+          // Here we only want to compare the "normalized" path, without any optional chaining ("?.") operator
+          // foo?.bar -> foo.bar
+          satisfyingPaths.add(path.replace(/\?$/, ''));
```

Comment:
The same change should be made below when adding path to missingPaths (line 1304). Version 4.04 is now getting an eslint error:
ESLint stack trace:
[Error - 4:20:28 PM] TypeError: Cannot read property 'references' of undefined

Your same change to missingPaths fixes that error.
```
missingPaths.add(path.replace(/\?$/, ''));
```

Reply:
Great catch! Just made this change.